### PR TITLE
Added Migration.md

### LINKED ISSUE #651: Migration Tutorial

### PR DESCRIPTION
**Pull request by eneufeld**
Fixes #651

### CODE REVIEW

File: MIGRATION.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+# Migrating from JSONForms 1 (AngluarJS)
+The complexity of the migration of an existing JSONForms 1 application, which is based on AngularJS, to JSONForms 2 depends on the feature set you use.
+
+## Main Features of JSONForms 2
+There are two big changes between JSONForms 1 and JSONForms 2 you need to understand when migrating your existing application.
+
+1. JSONForms 2 uses [redux](https://redux.js.org/) in order to maintain the state. Your application will need to add redux as a dependency.
```
Comment (edgarmueller) [inline MIGRATION.md]: Rephrase: Since JSON Forms 2 maintains its internal state via redux, you will need to add it as a dependency to your application.

File: MIGRATION.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+2. JSONForms 2 does not rely on any specific ui framework. The `2.0.0` release only provides renderers based on [React](https://reactjs.org) but we are working on an [Angular](https://angular.io) renderer-set for `2.1.0`.
```
Comment (edgarmueller) [inline MIGRATION.md]: ...on any specific UI framework [or library].
Rephrase 2nd sentence: The 2.0.0 initial release features renderers based on React. A Angular based renderer set will be be released with 2.1.0

File: MIGRATION.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+## Breaking changes for declarative files
+The UISchema for controls was simplified and the unneeded `ref` object inside `scope` was removed.
+So instead of:
```
Comment (edgarmueller) [inline MIGRATION.md]: Remove 'So'

File: MIGRATION.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+```ts
+const uischema = {
+    type: 'Control',
+    scope: {
+        $ref: '#/properties/name'
+    }
+}
+```
+the UISchema is now:
```
Comment (edgarmueller) [inline MIGRATION.md]: simply write

File: MIGRATION.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+```ts
+const uischema = {
+    type: 'Control',
+    scope: '#/properties/name'
+}
+```
+Otherwise all elements were kept and work as they used to in JSONForms 1.
+
+## Migrating Custom Renderers
+If you have custom renderers then you will have to reimplement them.
```
Comment (edgarmueller) [inline MIGRATION.md]: Rephrase: Any custom renderer need to be re-implemented. You can find instructions how to implement Custom controls based on React here.

File: MIGRATION.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+You can find the react based tutorial [here](http://jsonforms.io/docs/custom-renderers).
+
+## Using JSONForms 2
+As JSONForms 2 does not rely on any specific ui framework you can use it either nativily, as shown for React in the [Tutorial](http://jsonforms.io/docs/tutorial), or you can use the [WebComponent](https://www.webcomponents.org/introduction) that is provided.
```
Comment (edgarmueller) [inline MIGRATION.md]: UI framework or library
natively
Webcomponent wrapper

File: MIGRATION.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+## Migrating Custom Renderers
+If you have custom renderers then you will have to reimplement them.
+You can find the react based tutorial [here](http://jsonforms.io/docs/custom-renderers).
+
+## Using JSONForms 2
+As JSONForms 2 does not rely on any specific ui framework you can use it either nativily, as shown for React in the [Tutorial](http://jsonforms.io/docs/tutorial), or you can use the [WebComponent](https://www.webcomponents.org/introduction) that is provided. 
+
+1. Please first check the [Breaking changes for declarative files](#breaking-changes-for-declarative-files) before continuing.
+2. Now you can simply use the JSONForms 2 Webcomponent `json-forms` and pass in your data, JsonSchema and UISchema.
```
Comment (edgarmueller) [inline MIGRATION.md]: Remove 'simply' We're not passing in data and the schemata directly, maybe rephrase this part (as outlined below the listing).
Use correct casing for JSON Schema/UI schema.

**Comment by koegel** [comment]
@edgarmueller Can you please review again.

**Comment by lucas-koehler** [comment]
@koegel when taking a brief look at it, I noticed a typo in the first headline. There it says `AngluarJS` instead of `AngularJS`

**Comment by koegel** [comment]
OMG, true, I fixed it.

File: MIGRATION.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+    }
+);
+body.appendChild(jsonForms);
+```
+This snippet instantiates the JSON Forms WebComponent and passes in the redux store which consists of the jsonformsReducer and the initial state with the default Material renderers.
+
+Custom renderers can be added either through the initial state or registered explicitly as shown in the [Day 6 example](packages/examples/src/day6.ts).
+
+## Step 3: Migrate Custom Renderers
+Any custom renderer need to be re-factored to conform to the new custom renderer style in JSON Forms 2.x. You can find instructions how to implement Custom controls based on React [here](http://jsonforms.io/docs/custom-renderers). While you need to change a lot except for the template, the good news it that writing custom renderers became much simpler in JSON Forms 2 since the framework will trigger rendering and re-rendering in case of changes to the data or other state. In many cases this means you will be able to streamline your code for custom renderers significantly.
```
Comment (edgarmueller) [inline MIGRATION.md]: need[s]

File: MIGRATION.md [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+        jsonforms: jsonformsReducer()
+    }),
+    {
+        jsonforms: {
+            core: {
+                data,
+                schema,
+                uischema
+            },
+            materialRenderers,
```
Comment (edgarmueller) [inline MIGRATION.md]: keys are missing

Comment (eneufeld) [reply MIGRATION.md]: why do we need the keys? I thought that if the constant name equals the key we don't need the key.
